Source/RapyutaSimulationPlugins/Private/Robots/RobotVehicle.cpp

```diff
@@ -2,38 +2,123 @@
 
 #include "Robots/RobotVehicle.h"
 
-#include "Drives/RobotVehicleMovementComponent.h"
+// rclUE
 #include "Msgs/ROS2TFMsg.h"
 #include "ROS2Node.h"
 
-void ARobotVehicle::InitializeMoveComponent()
+// RapyutaSimulationPlugins
+#include "Drives/RobotVehicleMovementComponent.h"
+#include "RRROS2GameMode.h"
+#include "Tools/ROS2Spawnable.h"
+#include "Tools/SimulationState.h"
+
+ARobotVehicle::ARobotVehicle()
+{
+    Initialize();
+}
+
+ARobotVehicle::ARobotVehicle(const FObjectInitializer& ObjectInitializer) : Super(ObjectInitializer)
 {
-    RobotVehicleMoveComponent = NewObject<URobotVehicleMovementComponent>(this, TEXT("RobotVehicleMoveComponent"));
+    Initialize();
 }
 
-void ARobotVehicle::Tick(float DeltaSeconds)
+void ARobotVehicle::Initialize()
 {
-    Super::Tick(DeltaSeconds);
+    SkeletalMeshComp = CreateDefaultSubobject<USkeletalMeshComponent>(*FString::Printf(TEXT("%s_SkeletalComp"), *GetName()));
+    SkeletalMeshComp->VisibilityBasedAnimTickOption = EVisibilityBasedAnimTickOption::AlwaysTickPose;
+    SkeletalMeshComp->SetCollisionProfileName(UCollisionProfile::Pawn_ProfileName);
+    AddOwnedComponent(SkeletalMeshComp);
+    RootComponent = SkeletalMeshComp;
+}
+
+void ARobotVehicle::OnConstruction(const FTransform& InTransform)
+{
+    Super::OnConstruction(InTransform);
+
+    if (VehicleMoveComponentClass)
+    {
+        // (NOTE) Being created in [OnConstruction], PIE will cause this to be reset anyway, thus requires recreation
+        RobotVehicleMoveComponent = NewObject<URobotVehicleMovementComponent>(
+            this, VehicleMoveComponentClass, *FString::Printf(TEXT("%s_MoveComp"), *GetName()));
+        RobotVehicleMoveComponent->RegisterComponent();
+    }
+    else
+    {
+        // [OnConstruction] could run in various Editor BP actions, thus could not do Fatal log here
+        UE_LOG(LogRapyutaCore, Warning, TEXT("[%s] [VehicleMoveComponentClass] has not been configured!"), *GetName());
+    }
+}
+
+bool ARobotVehicle::InitSensors(AROS2Node* InROS2Node)

```

Comment:
I kinda wanted to name it more generically for all sensor types since this is quite a top-parent class, we could extract lidar init sources into InitLidars() then call it herein, when we have other types like depth.

Reply:
In that case this function should be virtual. then user can crate their own InitSensor based on this class.

Reply:
and maybe better to change lidarComponents to sensorComponents and use generic name(maybe better to create sensorBase class such as RobotVehicleMovecompoennt). 
there is a robot which do not have senr but camera

Reply:
Well, it is true but normally I'd avoid early generalization without specific needs yet. Basically, when we implement a new sensor type, we would go with steps you suggested, which should be done in a separate PR.